src/Groups/matrices/transform_form.jl

```diff
@@ -0,0 +1,412 @@
+      D1 = _elim_hyp_lines(A1)*D1
+      D2 = _elim_hyp_lines(A2)*D2
+      issquare( prod(diagonal(A1))*prod(diagonal(A2)) )[1] || return false, nothing
+      # move all the squares on the diagonal at the begin
+      _squares = [i for i in 1:n if issquare(A1[i,i])[1]]
+      our_perm = vcat(_squares, [i for i in 1:n if !(i in _squares)])      # TODO is there a more elengant way?
+      P = permutation_matrix(F,our_perm)
+      s1 = length(_squares)
+      D1 = P*D1
+      A1 = P*A1*transpose(P)
+      _squares = [i for i in 1:n if issquare(A2[i,i])[1]]
+      our_perm = vcat(_squares, [i for i in 1:n if !(i in _squares)])      # TODO is there a more elengant way?
+      P = permutation_matrix(F,our_perm)
+      s2 = length(_squares)
+      D2 = P*D2
+      A2 = P*A2*transpose(P)
+      # get same number of squares on the two diagonals of A1 and A2 by modifying A1
+      if s1!=s2
+         s = min(s1,s2)+1
+         w = A1[s,s]*A2[s,s]     # I'm sure this is not a square
+         a,b = _solve_eqn(F(1),F(1),w)
+         L = identity_matrix(F,n)
+         for i in 0:div(abs(s1-s2),2)-1
+            L[s+2*i,s+2*i] = a
+            L[s+2*i,s+2*i+1] = b*square_root(A1[s+2*i,s+2*i]*A1[s+2*i+1,s+2*i+1]^-1)
+            L[s+2*i+1,s+2*i] = b
+            L[s+2*i+1,s+2*i+1] = -a*square_root(A1[s+2*i,s+2*i]*A1[s+2*i+1,s+2*i+1]^-1)
+         end
+         D1 = L*D1
+         A1 = L*A1*transpose(L)
+      end
+      # change matrix from A1 to A2
+      S = diagonal_matrix([square_root(A1[i,i]*A2[i,i]^-1) for i in 1:n])
+      return true, D1^-1*S*D2
+   end
+
+end
+
+"""
+    iscongruent(f::SesquilinearForm{T}, g::SesquilinearForm{T}) where T <: RingElem
+If `f` and `g` are sesquilinear forms, return (`true`, `C`) if there exists a matrix `C` such that `f^C = g`, or equivalently, `CBC* = A`, where `A` and `B` are the Gram matrices of `f` and `g` respectively, and `C*` is the transpose-conjugate matrix of `C`. If such `C` does not exist, then return (`false`, `nothing`).
+If `f` and `g` are quadratic forms, return (`true`, `C`) if there exists a matrix `C` such that `f^A = ag` for some scalar `a`. If such `C` does not exist, then return (`false`, `nothing`).
+"""
+function iscongruent(f::SesquilinearForm{T}, g::SesquilinearForm{T}) where T <: RingElem
+
+   @assert base_ring(f)==base_ring(g) "The forms have not the same base ring"
+   @assert nrows(gram_matrix(f))==nrows(gram_matrix(g)) "The forms act on vector spaces of different dimensions"
+   f.descr==g.descr || return false, nothing
+   n = nrows(gram_matrix(f))
+   F = base_ring(f)
+   
+   if f.descr==:quadratic
+      if iseven(characteristic(F))            # in this case we use the GAP algorithms
+         Bg = g.mat_iso(GAP.Globals.BaseChangeToCanonical(g.X))
+         Bf = f.mat_iso(GAP.Globals.BaseChangeToCanonical(f.X))
+         if _is_scalar_multiple_mat(_upper_triangular_version(Bf*gram_matrix(f)*transpose(Bf)),_upper_triangular_version(Bg*gram_matrix(g)*transpose(Bg)))[1]
```

Comment:
Let's try to not do too much in one line... IMHO the following is far more readable
```suggestion
         Uf = _upper_triangular_version(Bf*gram_matrix(f)*transpose(Bf))
         Ug = _upper_triangular_version(Bg*gram_matrix(g)*transpose(Bg))
         if _is_scalar_multiple_mat(Uf,Ug)[1]
```